### ISSUE TITLE
How can I eval this simple model to make a prediction ?

### ISSUE THREAD

**Issue by hvaneylen** [Microsoft/CNTK#3235]
I want to have a simple kind of "hello world" cntk model that I master and I'm stuck.
Here is the code - can anybody tell me how to eval a simple prediction. This is a 2 features / 3 classes classification problem with a dense + softmax model. I'm not very skilled in python and cntk so I surely miss something you will find simple.
Thx for any help

from matplotlib import pyplot
from pandas import DataFrame
import numpy as np

import cntk as C

from sklearn.preprocessing import OneHotEncoder

# générer un jeu de données avec 2 caractéristiques et
# trois classes
X,y = make_blobs(n_samples=180, centers=3, n_features=2, center_box=(0,10), cluster_std=1.0)
X = X.astype(np.float32)
y = y.astype(np.float32)

# range les données dans un dataframe de (100, 3)
df = DataFrame(dict(x1=X[:,0],x2=X[:,1], label=y ))
print(df.shape)
colors={0:'red', 1:"blue",2:"green"}
fig, ax = pyplot.subplots()
grouped = df.groupby("label")
print(grouped.describe())
for key, group in grouped:
    group.plot(ax=ax, kind="scatter", x="x1", y="x2", label=key, color=colors[key])

x1_min = df.x1.min()
x1_max = df.x1.max()
x2_min = df.x2.min()
x2_max = df.x2.max()

print("x1 min, max :[{:0.2f},{:0.2f}] ".format( x1_min, x1_max))
print("x2 min, max :[{:0.2f},{:0.2f}] ".format( x2_min, x2_max))

# crée un point à classer
t1 = np.random.uniform(x1_min, x1_max)
t2 = np.random.uniform(x2_min, x2_max)
print("t1 :{:0.2f}, t2 :{:0.2f}".format(t1,t2))
pyplot.plot(t1,t2,"x")

pyplot.show()

X_train = X[:120]
X_test = X[121:]
print(X_train[:5])
print(X_test[:5])

# encode en onehot les trois catégories
onehot_encoder = OneHotEncoder(sparse=False)
Y_train = onehot_encoder.fit_transform(y[:120].reshape(-1,1))
Y_test = onehot_encoder.fit_transform(y[121:].reshape(-1,1))
# nécessaire de transformer en float32 pour être mis en oeuvre par cntk
Y_train = Y_train.astype(np.float32)
Y_test = Y_test.astype(np.float32)

print("data = \n", X_train[:4])
print("labels =\n", Y_train[:4])

# création du modèle de régression logistique
## Define a small Logistic Regression model function
# 2 caractéristiques en entrées
x = C.input_variable(2)
# 3 labels en sorties
y = C.input_variable(3, is_sparse=False)
# une seule couche Dense
model = C.layers.Dense(3, activation=None)
# un softmax en sortie
loss = C.cross_entropy_with_softmax(model(x), y)
print(loss)

## entraînement
learner = C.sgd(model.parameters,
                C.learning_parameter_schedule(0.1))
progress_writer = C.logging.ProgressPrinter(0)

train_summary = loss.train((X_train, Y_train), parameter_learners=[learner],
                   callbacks=[progress_writer])

# predictions
features = np.asarray([t1,t2], dtype=np.float32)
print("loss.arguments[0]: ", loss.arguments[0])
# this doent work - how to do the eval ?
result = loss.eval({x:features})  # loss.arguments[0] doesnt work too
print("result : " , result)

**Comment by jaliyae** [comment]
You need two corrections to the above code. 

First cross_entropy_with_softmax calculates the entropy between two vectors, which require two inputs. 
Please refer to [this ](https://cntk.ai/pythondocs/cntk.losses.html)page for more details.

For evaluation, you can use softmax like this.
```
out = C.softmax(model)
result = out.eval({x:features}) 
print("result : " , result)
```

Next, in the line where you use the layers library does not bind the input variable to the model.
`loss = C.cross_entropy_with_softmax(model(x), y)`

Instead, please use the following.
```
model = C.layers.Dense(3, activation=None)(x)
loss = C.cross_entropy_with_softmax(model, y)
```
With those changes, I was able to get it running.  Please refer to [CNTK tutorials](https://cntk.ai/pythondocs/tutorials.html) for more examples.

Once you create a model, it is easier to see how it looks and the shapes of the inputs but plotting it like this.
`C.logging.graph.plot(out, "dense_z.pdf")`

**Comment by hvaneylen** [comment]
Your corrections work fine - thx a lot
